src/batch/client.ts

```diff
@@ -131,33 +130,22 @@ export class BatchTranscription {
       throw 'Error: submitResponse is undefined';
     }
 
-    await waitUntilAvail(
+    await poll(
       async () => {
         const { job } = await this.getJob(submitResponse.id);
         if (job.status === 'rejected') {
           throw job.errors;
         } else if (job.status === 'done') {
-          return true;
+          return { state: 'resolved', value: job };
         } else {
-          return false;
+          return { state: 'pending' };
```

Comment:
The result from poll isn't getting assigned to anything, so is `value: job` irrelevant here?

If we're not using `value` elsewhere, perhaps returning true or false is sufficient and less complex?

Reply:
I see you've added tests that use the return value, which is a nice feature, but if we're not using it in the SDK code perhaps we should remove this for now (and add it back in if and when we need it later)

Reply:
Yeah I guess I was just trying to make something abstract enough to be useful for more than just waiting, but perhaps we don't need it, I don't know

Reply:
Do you mean returning true/false in this call site, or removing the `PollResult` type alrogether?

Reply:
I would suggest we remove PollResult and replace it with a boolean for now, unless we need to use any of the features that PollResult provides.

I could see it being useful, but if we're not using it I would personally remove it.

I don't mind if you foresee using it and want to keep it, just a suggestion.

Reply:
another alternative would be for `PollResult` and `boolean` to be a valid return type, but that's really if we're creating an abstract util which I don't think is what's needed here